### ISSUE TITLE
Does not create groups automatically with name "Booking name - Option name"

### ISSUE THREAD

**Issue by demontech** [Wunderbyte-GmbH/moodle-mod_booking#71]
Hi there,

I'm setting up to automatically enroll users in the course and enroll users in group both to Yes. The first setting works, but it doesn't create groups. 

![image](https://cloud.githubusercontent.com/assets/8148238/21376657/ba8967b4-c6eb-11e6-8341-8d25b889f7e5.png)

As matter of fact, when one student changed the status from waitlisted to booked (student 6 in picture) that's when it created the group.

I tested with v 2.3 and 2.4 with moodle 3.0 and 3.1 ..... same issue. Also on a new moodle instance with booking as the only plugin. My version of PHP 5.6.27 with MySQL 5.6.32

Thanks!

**Comment by gemguardian** [comment]
hello, 
I am working on Moodle 3.1 and the plug-in doesn't seem to create any groups anymore - same as mentioned above. Though I haven't tested it with the waitlisted booking, since i don't use that feature in the booking. 

I have tested the Module before and I am sure I have seen it work on Moodle 3.1 but in an earlier version (3.1?). But since I have an upgrade to 3.1.4 it does seem to enrol students into a course, but not creat a group and enrol students in a group. 

I have tested on a local 3.1.2(build 20160912 / MySql 5.6.33 and PHP 5.6.26) installation, and I do see the group creation but the enrollment isn't happening. I am not sure if the group enrolment is forced by a cron action, I can't test a cron force on my local machine. 

I did test the plug-in on 2 Moodle 3.1.4 enviroments that are hosted by seperate hosting parties, on different PHP and SQL versions. And since it doesn't work on both, and reading above, I thought I would mention it here. 

Moodle 3.1.4+(build 20170112): PHP 5.6.29 with MySql 5.5.31
Moodle 3.1.4 (build 20170109): PHP 7.0.8.0.0.16.04.3 / MySQL 5.5.52.0.0.4.04.1

I think that some change between Moodle 3.1.2 and 3.1.4 has effected the working of this plug-in. 

Thank you
Gemma 

-
Moodle 3.1.4 (build 20170109) give me the folowing debugging notifications:

Method tag_get_tags_array() is deprecated and replaced with core_tag_tag::get_item_tags_array(). Component is now required when retrieving tag instances.
line 2631 of /lib/deprecatedlib.php: call to debugging()
line 421 of /mod/booking/mod_form.php: call to tag_get_tags_array()
line 470 of /course/moodleform_mod.php: call to mod_booking_mod_form->data_preprocessing()
line 256 of /course/modedit.php: call to moodleform_mod->set_data()

Notice: Undefined variable: CFG in /data/www/moodle.ltnc.nl/public_html/mod/booking/lib.php on line 279

Notice: Trying to get property of non-object in /data/www/moodle.ltnc.nl/public_html/mod/booking/lib.php on line 279
Function tag_set() is deprecated. Use core_tag_tag::set_item_tags() instead
line 3470 of /lib/deprecatedlib.php: call to debugging()
line 282 of /mod/booking/lib.php: call to tag_set()
line 571 of /course/modlib.php: call to booking_update_instance()
line 273 of /course/modedit.php: call to update_moduleinfo()
